drv/lpc55-spi-server/src/main.rs

```diff
@@ -63,74 +369,339 @@ fn main() -> ! {
-
-    let mut a_bytes: [u8; 8] = [0xaa; 8];
-    let mut b_bytes: [u8; 8] = [0; 8];
-
     sys_irq_control(1, true);
 
-    let mut tx = &mut a_bytes;
-    let mut rx = &mut b_bytes;
-
-    let mut tx_cnt = 0;
-    let mut rx_cnt = 0;
+    // Configure ROT_IRQ
+    let rot_irq = Pin::PIO0_18; // XXX Should be in app.toml
+                                // Direction must be set after other pin configuration.
+    gpio.set_dir(rot_irq, Direction::Output).unwrap_lite();
+    // At the begining, we are ready to receive and have nothing to transmit.
+    // Ensure that ROT_IRQ is not asserted
+    gpio.set_val(rot_irq, Value::One).unwrap_lite();
+    ringbuf_entry!(Trace::DeassertRotIrq);
+
+    // Configure SP_RESET
+    // Reset is an input to detect SP internal reset but can be an driven low
+    // to reset the SP.
+
+    let sp_reset = Pin::PIO0_9; // XXX Should be in app.toml
+    gpio.set_dir(sp_reset, Direction::Input).unwrap_lite();
+    // TODO: SP reset and reset detection don't have to be here,
+    // However, SP reset will affect Sprockets state, so here may be good.
+    // TODO: Detect SP reset events and invalidate any trust/session that may exist.
+    // TODO: Drive SP reset when appropriate.
+
+    let mut tx = [0u8; SPI_RSP_BUF_SIZE];
+    let mut tctx = &mut TxContext::new(&mut tx[..]);
+    let mut rx = [0u8; SPI_REQ_BUF_SIZE];
+    let mut rctx = &mut RxContext::new(&mut rx[..]);
 
-    spi.drain();
     spi.enable_rx();
-    spi.enable_tx();
-
-    let mut tx_done = false;
-    let mut rx_done = false;
+    spi.disable_tx(); // Ignore Tx interrupts for now.
+    spi.send_u8(0x00); // Throw away data, but this sets bit-level framing.
+    spi.drain(); // Nothing in the FIFOs is interesting yet.
+    spi.rxerr_clear();
+    // spi.txerr_clear();
+
+    // Take interrupts on CSn asserted and de-asserted.
+    // The assert interrupt happens before the first byte of Rx data is
+    // available. That could make us more reponsive if keeping up with realtime
+    // requirements becomes an issue.
+    // On de-assert, we'll process any received message and then queue up a
+    // response for the next CSn assert.
+    spi.ssa_enable(); // Interrupt on CSn changing to asserted.
+    spi.ssd_enable(); // Interrupt on CSn changing to deasserted.
+    ringbuf_entry!(Trace::Initialized); // XXX
+
+    let mut olc = 0u16;
+
+    // Collect info for status
+    // TODO: Should the status data returned be in some extensible format?
+    // Information should include items needed before a system is fully configured
+    // or other cases where the RoT may not be in a compliant state.
+    // The information should guide the next steps for
+    //   - RoT FW version: used to assess the need for FW update.
+    //   - Boot ROM CRC: a proxy for machine readable NXP LPC55 revision.
+    pub const CRC32: Crc<u32> = Crc::<u32>::new(&CRC_32_CKSUM);
+    let bootrom_crc32 = CRC32.checksum(&bootrom().data[..]).to_le_bytes();
+
+    let (
```

Comment:
I already said this, but this would be great as a `struct LStat`. Plus, then you could `#[derive(Eq, PartialEq)]` and not have to compare all of the values individually down below!

Reply:
Done